src/AssetsPage/AssetsTable/AssetsTable.test.jsx

```diff
@@ -24,31 +28,335 @@ const defaultProps = {
+  describe('renders', () => {
+    beforeEach(() => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+    });
+    it(' correct number of assets', () => {
+      // + 1 because it includes row item for headings
+      expect(wrapper.find('tr')).toHaveLength(defaultProps.assetsList.length + 1);
+    });
+    it('correct number of trashcan buttons', () => {
+      expect(wrapper.find('button').filterWhere(button => button.hasClass('fa-trash'))).toHaveLength(defaultProps.assetsList.length);
+    });
+  });
+  describe('onSortClick', () => {
+    let updateSortSpy;
+
+    beforeEach(() => {
+      updateSortSpy = jest.fn();
+    });
+
+    const sortableAssetProps = [
+      {
+        sort: 'date_added',
+        direction: 'desc',
+      },
+      {
+        sort: 'date_added',
+        direction: 'asc',
+      },
+      {
+        sort: 'date_added',
+        direction: 'desc',
+      },
+    ];
+
+    const assertValues = [
+      {
+        buttonIndexToClick: 2,
+        buttonToClickText: 'Date Added',
+        onSortColumnParameter: 'date_added',
+        onSortDirectionParameter: 'asc',
+      },
+      {
+        buttonIndexToClick: 2,
+        buttonToClickText: 'Date Added',
+        onSortColumnParameter: 'date_added',
+        onSortDirectionParameter: 'desc',
+      },
+      {
+        buttonIndexToClick: 0,
+        buttonToClickText: 'Name',
+        onSortColumnParameter: 'display_name',
+        onSortDirectionParameter: 'desc',
+      },
+    ];
+
+    sortableAssetProps.forEach((sortableAssetProp, index) => {
+      it(`calls onSort function with ${sortableAssetProp.direction} click on ${sortableAssetProp.sort === 'date_added' ? 'same' : 'different'} column`, () => {
+        const sortableAssetParameters = {
+          ...defaultProps.assetsParameters,
+          ...sortableAssetProp,
+        };
+
+        const testValues = assertValues[index];
+
+        wrapper = mount(
+          <AssetsTable
+            {...defaultProps}
+            updateSort={updateSortSpy}
+            assetsParameters={sortableAssetParameters}
+          />,
+        );
+
+        const sortButtons = wrapper.find('button').filterWhere(button => (
+          button.find('span > span .fa-sort').exists() ||
+          button.find('span > span .fa-sort-desc').exists() ||
+          button.find('span > span .fa-sort-asc').exists()
+        ));
+
+        expect(sortButtons).toHaveLength(3);
+
+        expect(sortButtons.at(testValues.buttonIndexToClick).filterWhere(button => (
+          button.text().includes(testValues.buttonToClickText)
+        ))).toHaveLength(1);
+
+        sortButtons.at(testValues.buttonIndexToClick).simulate('click');
+
+        expect(updateSortSpy).toHaveBeenCalledTimes(1);
+        expect(updateSortSpy).toHaveBeenLastCalledWith(
+          testValues.onSortColumnParameter,
+          testValues.onSortDirectionParameter,
+        );
+      });
+    });
+  });
```

Comment:
Nice test - I was able to understand it clearly when looking at the updated file as a whole.  It also covers a lot of different testing features that we should leverage when creating future tests.

Reply:
Do you think there's a way it could be clearer? I see that the sortableAssetProps might have been a little confusing.

Reply:
I think it's fine, seeing it as a whole makes sense.  Wouldn't worry about it too much at this point.